### ISSUE TITLE
Can fields have reference to resource instead of just the data?

### ISSUE THREAD

**Issue by jeffreyguenther** [thoughtbot/administrate#438]
For example, I'd like to add a password field.

``` ruby
class PasswordField < Administrate::Field::Base
  def to_s
    if data.present?
        "*" * data.size
    else
        "Not set"
    end
  end
end
```

This makes sense until you realize that `data` is referring to the string provided by `password`. If you're using a library like `clearance` you don't have a password attribute on your user model. You have `encrypted_password`. I want to detect if `encrypted_password` has any content and format the string accordingly.

**Comment by straight-shoota** [comment]
I'd definitly like this as well and I don't think, there would be any harm in that. Every instance of a field class is already specific to a single resource by holding its distinct attribute value. Why not make that resource directly available from the instance?
There are a lot of good reasons for access to the resource or other attributes.
The autosafe feature for [SimpleMDE](https://github.com/NextStepWebs/simplemde-markdown-editor) for example needs a unique identifier, for which I'd want to use the resource class and record id. In the template, there is already access to the record through the FormBuilder (`f.object`). But this is only available in `form` views, not in index or show. Genereally it would be better to put domain logic into the field class, not the template.
It should be quite simple to extend `Page::Base#attribute_field` and `Field::Base#initialize` to deliver the current resource to the field instance.

**Comment by mcmire** [comment]
Also see #267.

**Comment by nickcharlton** [comment]
I _think_ this issue was solved with #706, so I'm going to close this. But please open a new one if that's not the case.
